src/main/java/reactor/rabbitmq/Sender.java

```diff
@@ -472,21 +494,39 @@ public RpcClient rpcClient(String exchange, String routingKey, Supplier<String>
     }
 
     public void close() {
-        if (hasConnection.getAndSet(false)) {
-            try {
-                // FIXME use timeout on block (should be a parameter of the Sender)
-                connectionMono.block().close();
-            } catch (IOException e) {
-                throw new RabbitFluxException(e);
+        if (closingOrClosed.compareAndSet(false, true)) {
+            if (hasConnection.getAndSet(false)) {
+                safelyExecute(
+                        () -> connectionMono.block().close(this.connectionClosingTimeout),
+                        "Error while closing sender connection"
+                );
             }
+
+            if (this.privateConnectionSubscriptionScheduler) {
+                safelyExecute(
+                        () -> this.connectionSubscriptionScheduler.dispose(),
+                        "Error while disposing connection subscription scheduler"
+                );
+            }
+            if (this.privateResourceManagementScheduler) {
+                safelyExecute(
+                        () -> this.resourceManagementScheduler.dispose(),
+                        "Error while disposing resource management scheduler"
+                );
+            }
+            safelyExecute(
+                    () -> channelCloseThreadPool.shutdown(),
+                    "Error while closing channel closing thread pool"
+            );
         }
-        if (this.privateConnectionSubscriptionScheduler) {
-            this.connectionSubscriptionScheduler.dispose();
-        }
-        if (this.privateResourceManagementScheduler) {
-            this.resourceManagementScheduler.dispose();
+    }
+
+    private void safelyExecute(Utils.ExceptionRunnable action, String message) {
+        try {
+            action.run();
+        } catch (Exception e) {
+            LOGGER.warn(message + ": {}", e.getCause());
```

Comment:
Same suggestion as in Receiver.safelyExecute

Reply:
Same as above.